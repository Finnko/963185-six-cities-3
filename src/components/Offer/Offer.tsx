import React from 'react';

import { IOffer } from '../../interfaces/offer';
import { getRatingPercentage } from '../../utils/common';

interface OfferProps {
  offer: IOffer;
}

const Offer: React.FC<OfferProps> = ({ offer }) => {
  const {
    title,
    type,
    price,
    isPremium,
    rating,
    bedrooms,
    maxAdults,
    goods,
    description,
    host: {
      name,
      avatarUrl,
      isPro
    }
  } = offer;

  const premium = isPremium ? (
    <div className="property__mark">
      <span>Premium</span>
    </div>) : null;

  const ratingPercent = getRatingPercentage(rating);

  return (
    <>
      {premium}
      <div className="property__name-wrapper">
        <h1 className="property__name">
          {title}
        </h1>
        <button className="property__bookmark-button button" type="button">
          <svg className="property__bookmark-icon" width="31" height="33">
            <use xlinkHref="#icon-bookmark"/>
          </svg>
          <span className="visually-hidden">To bookmarks</span>
        </button>
      </div>
      <div className="property__rating rating">
        <div className="property__stars rating__stars">
          <span style={{width: `${ratingPercent}%`}} />
          <span className="visually-hidden">Rating</span>
        </div>
        <span className="property__rating-value rating__value">{rating}</span>
      </div>
      <ul className="property__features">
        <li className="property__feature property__feature--entire">
          {type}
        </li>
        <li className="property__feature property__feature--bedrooms">
          {bedrooms} Bedrooms
        </li>
        <li className="property__feature property__feature--adults">
          Max {maxAdults} adults
        </li>
      </ul>
      <div className="property__price">
        <b className="property__price-value">&euro;{price}</b>
        <span className="property__price-text">&nbsp;night</span>
      </div>
      <div className="property__inside">
        <h2 className="property__inside-title">What&apos;s inside</h2>
        <ul className="property__inside-list">
          {goods.map((good, index) => (
            <li key={good + index} className="property__inside-item">{good}</li>
          ))}
        </ul>
      </div>
      <div className="property__host">
        <h2 className="property__host-title">Meet the host</h2>
        <div className="property__host-user user">
          <div className={`property__avatar-wrapper user__avatar-wrapper ${isPro ? 'property__avatar-wrapper--pro' : ''} `}>
            <img className="property__avatar user__avatar" src={`/${avatarUrl}`} width="74" height="74" alt="Host avatar"/>
          </div>
          <span className="property__user-name">
            {name}
          </span>
        </div>
        <div className="property__description">
          <p className="property__text">
            {description}
          </p>
        </div>
      </div>
    </>
  );
};

export default Offer;
