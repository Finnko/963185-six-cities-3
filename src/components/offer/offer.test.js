import React from "react";
import renderer from "react-test-renderer";
import Offer from './offer';

const mockDate = new Date(1583591483969).valueOf();
const mock = {
  id: `id_11`,
  title: `Beautiful & luxurious apartment at great location`,
  price: `120`,
  type: `Apartment`,
  promoImage: `img/apartment-01.jpg`,
  isPremium: true,
  rating: 2.5,
  images: [
    `img/room.jpg`,
    `img/apartment-01.jpg`,
    `img/apartment-01.jpg`,
  ],
  roomsCount: 2,
  guestsCount: 4,
  features: [`Wi-Fi`, `Baby seat`, `Cabel TV`, `Fridge`],
  owner: {
    name: `Dominika`,
    avatar: `img/avatar-angelina.jpg`,
    description: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`],
    isPro: true
  },
  reviews: [
    {
      id: 1,
      name: `Max`,
      avatar: `/img/avatar-max.jpg`,
      rating: 1.5,
      description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      date: mockDate
    },
    {
      id: 2,
      name: `Alexey`,
      avatar: `/img/avatar-max.jpg`,
      rating: 0.5,
      description: `The building is green and from 18th century.`,
      date: mockDate
    },
    {
      id: 3,
      name: `Anastasia`,
      avatar: `/img/avatar-max.jpg`,
      rating: 4.5,
      description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      date: mockDate
    }]
};


it(`Should Offer component render correctly`, () => {
  const tree = renderer
    .create(<Offer
      offer={mock}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
