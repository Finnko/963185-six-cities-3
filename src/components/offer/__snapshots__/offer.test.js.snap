// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Offer component render correctly 1`] = `
Array [
  <div
    className="property__mark"
  >
    <span>
      Premium
    </span>
  </div>,
  <div
    className="property__name-wrapper"
  >
    <h1
      className="property__name"
    >
      Beautiful & luxurious apartment at great location
    </h1>
    <button
      className="property__bookmark-button button"
      type="button"
    >
      <svg
        className="property__bookmark-icon"
        height="33"
        width="31"
      >
        <use
          xlinkHref="#icon-bookmark"
        />
      </svg>
      <span
        className="visually-hidden"
      >
        To bookmarks
      </span>
    </button>
  </div>,
  <div
    className="property__rating rating"
  >
    <div
      className="property__stars rating__stars"
    >
      <span
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <span
        className="visually-hidden"
      >
        Rating
      </span>
    </div>
    <span
      className="property__rating-value rating__value"
    >
      2.5
    </span>
  </div>,
  <ul
    className="property__features"
  >
    <li
      className="property__feature property__feature--entire"
    >
      Apartment
    </li>
    <li
      className="property__feature property__feature--bedrooms"
    >
      2
       Bedrooms
    </li>
    <li
      className="property__feature property__feature--adults"
    >
      Max 
      4
       adults
    </li>
  </ul>,
  <div
    className="property__price"
  >
    <b
      className="property__price-value"
    >
      €
      120
    </b>
    <span
      className="property__price-text"
    >
       night
    </span>
  </div>,
  <div
    className="property__inside"
  >
    <h2
      className="property__inside-title"
    >
      What's inside
    </h2>
    <ul
      className="property__inside-list"
    >
      <li
        className="property__inside-item"
      >
        Wi-Fi
      </li>
      <li
        className="property__inside-item"
      >
        Baby seat
      </li>
      <li
        className="property__inside-item"
      >
        Cabel TV
      </li>
      <li
        className="property__inside-item"
      >
        Fridge
      </li>
    </ul>
  </div>,
  <div
    className="property__host"
  >
    <h2
      className="property__host-title"
    >
      Meet the host
    </h2>
    <div
      className="property__host-user user"
    >
      <div
        className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro "
      >
        <img
          alt="Host avatar"
          className="property__avatar user__avatar"
          height="74"
          src="img/avatar-angelina.jpg"
          width="74"
        />
      </div>
      <span
        className="property__user-name"
      >
        Dominika
      </span>
    </div>
    <div
      className="property__description"
    >
      <p
        className="property__text"
      >
        A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
      </p>
    </div>
  </div>,
]
`;
